### PR TITLE
Service Participant, Transport Registry, RTPS libs updated to support certain Restart scenarios

### PR DESCRIPTION
**Pull request by mitza-oci**
Related to #1005 and possibly #991 

- Support restarting the same Service_Participant object after shutdown() 
- Updated RtpsUdpLoader to have an easy way to reload the transport (from application)
- Sedp: remove transport when we're done 
- TransportRegistry::remove_inst() calls shutdown on the transport

### CODE REVIEW

File: tests/DCPS/Restart/RestartTest.cpp [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    participant_ = dpf->create_participant(0, PARTICIPANT_QOS_DEFAULT, 0, 0);
+    TheTransportRegistry->bind_config(cfg, participant_);
+    participant_->enable();
+
+    DDS::TypeSupport_var ts = new Messenger::MessageTypeSupportImpl;
+    CORBA::String_var tn = ts->get_type_name();
+    ts->register_type(participant_, tn);
+    DDS::Topic_var topic = participant_->create_topic("Topic", tn, TOPIC_QOS_DEFAULT, 0, 0);
+
+    DDS::Publisher_var pub = participant_->create_publisher(PUBLISHER_QOS_DEFAULT, 0, 0);
```
Comment (jrw972) [inline tests/DCPS/Restart/RestartTest.cpp]: Would it be valuable to have the publishers send messages as part of the test?
Should the test check for association?

File: tests/DCPS/Restart/RestartTest.cpp [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  {
+    Application a1;
+    {
+      Application a2;
+    }
+    {
+      Application a3;
+    }
+  }
+  Application a4;
```
Comment (jrw972) [inline tests/DCPS/Restart/RestartTest.cpp]: Would it be valuable to insert some sleeps to allow timers and other events to fire before exercising the shutdown code?

File: tests/DCPS/Restart/RestartTest.cpp (hunk 1 above)
Comment (mitza-oci) [reply tests/DCPS/Restart/RestartTest.cpp]: That was not necessary to reproduce the problem

File: tests/DCPS/Restart/RestartTest.cpp (hunk 2 above)
Comment (mitza-oci) [reply tests/DCPS/Restart/RestartTest.cpp]: In this case we're exercising the less-common path of getting to shutdown very quickly.

**Comment by dczanella** [comment]
This merge fixed the issue:  [Proposal test to use CORBA to create/destroy participant #991](https://github.com/objectcomputing/OpenDDS/pull/991).
How can I close it?

**Pull request merged**